pkg/blockbuilder/blockbuilder_test.go

```diff
@@ -408,7 +410,9 @@ func TestBlockBuilder_StartupWithExistingCommit(t *testing.T) {
 
 	opts := []kgo.Opt{
 		kgo.ClientID("1"), kgo.SeedBrokers(addr), kgo.ConsumeTopics(testTopic),
-		kgo.ConsumerGroup(testGroup), kgo.DisableAutoCommit(),
+		kgo.ConsumerGroup(testGroup),
+		kgo.Balancers(kgo.RoundRobinBalancer()),
```

Comment:
@codesome this one is what caused your issues with partitions not being assigned (mentioned in #8383). Kafka clients within the group must agree on the balancing strategy, otherwise you get:

```
INCONSISTENT_GROUP_PROTOCOL: The group member's supported protocols are incompatible with those of existing members or first group member tried to join with empty protocol type or empty protocol list.
```